### PR TITLE
docs: add missing Setting annotation with description to setting prop…

### PR DESCRIPTION
**Pull request by majadlymhmd**
…erties

## What this PR changes/adds

Add missing `@Setting` to Setting properties with description 
## Why it does that

improves the code's readability and maintainability
## Linked Issue(s)
ref #2710

## Checklist

- [ ] added appropriate tests?
- [ ] performed checkstyle check locally?
- [ ] added/updated copyright headers?
- [ ] documented public classes/methods?
- [ ] added/updated relevant documentation?
- [ ] assigned appropriate label? (exclude from changelog with label `no-changelog`)
- [ ] formatted title correctly? (_take a look at the [CONTRIBUTING](https://github.com/eclipse-edc/Connector/blob/main/CONTRIBUTING.md#submit-a-pull-request) and [Etiquette for pull requests](https://github.com/eclipse-edc/Connector/blob/main/pr_etiquette.md) for details_)

### CODE REVIEW

File: data-protocols/dsp/dsp-api-configuration/src/main/java/org/eclipse/edc/protocol/dsp/api/configuration/DspApiConfigurationExtension.java [hunk 1]
```diff
@@ -37,25 +38,23 @@
...
 @Provides({ DspApiConfiguration.class })
 public class DspApiConfigurationExtension implements ServiceExtension {
-    
+
     public static final String NAME = "Dataspace Protocol API Configuration Extension";
-    
     public static final String CONTEXT_ALIAS = "protocol";
-    
     public static final String DEFAULT_DSP_CALLBACK_ADDRESS = "http://localhost:8282/api/v1/dsp";
+    @Setting(value = "Callback URL address fot the DSP")
```
Comment (paullatzelsperger) [inline data-protocols/dsp/dsp-api-configuration/src/main/java/org/eclipse/edc/protocol/dsp/api/configuration/DspApiConfigurationExtension.java:45]: pls improve

File: extensions/control-plane/store/sql/contract-negotiation-store-sql/src/main/java/org/eclipse/edc/connector/store/sql/contractnegotiation/SqlContractNegotiationStoreExtension.java [hunk 2]
```diff
@@ -33,9 +34,9 @@
 @Extension(value = "SQL contract negotiation store")
 public class SqlContractNegotiationStoreExtension implements ServiceExtension {
 
+    @Setting(value = "Name of the data source for contract negotiation")
```
Comment (paullatzelsperger) [inline extensions/control-plane/store/sql/contract-negotiation-store-sql/src/main/java/org/eclipse/edc/connector/store/sql/contractnegotiation/SqlContractNegotiationStoreExtension.java:37]: pls improve

File: system-tests/runtimes/file-transfer-provider/src/main/java/org/eclipse/edc/test/extension/api/FileTransferExtension.java [hunk 3]
```diff
@@ -38,6 +39,7 @@
 public class FileTransferExtension implements ServiceExtension {
 
     public static final String USE_POLICY = "use-eu";
+    @Setting(value = "Path to test assets")
```
Comment (paullatzelsperger) [inline system-tests/runtimes/file-transfer-provider/src/main/java/org/eclipse/edc/test/extension/api/FileTransferExtension.java:42]: pls improve

**Review by paullatzelsperger — CHANGES_REQUESTED** [review]
unfortunately it is not sufficient to paraphrase the name of the constant. Please actually document what the setting is meant for, what it does, etc.
For example, I am 100% certain that documenting `DATASOURCE_NAME_SETTING` without the words "database", "jdbc" or "sql" is virtually impossible.

File: data-protocols/dsp/dsp-api-configuration/src/main/java/org/eclipse/edc/protocol/dsp/api/configuration/DspApiConfigurationExtension.java (hunk 1 above)
Comment (ndr-brt) [reply data-protocols/dsp/dsp-api-configuration/src/main/java/org/eclipse/edc/protocol/dsp/api/configuration/DspApiConfigurationExtension.java:45]: please add the `defaultValue` property as well

File: extensions/control-plane/store/sql/contract-negotiation-store-sql/src/main/java/org/eclipse/edc/connector/store/sql/contractnegotiation/SqlContractNegotiationStoreExtension.java (hunk 2 above)
Comment (ndr-brt) [reply extensions/control-plane/store/sql/contract-negotiation-store-sql/src/main/java/org/eclipse/edc/connector/store/sql/contractnegotiation/SqlContractNegotiationStoreExtension.java:37]: please add the `defaultValue` property as well

File: system-tests/runtimes/file-transfer-provider/src/main/java/org/eclipse/edc/test/extension/api/FileTransferExtension.java (hunk 3 above)
Comment (ndr-brt) [reply system-tests/runtimes/file-transfer-provider/src/main/java/org/eclipse/edc/test/extension/api/FileTransferExtension.java:42]: this is a `system-tests` runtime, I think we can avoid documenting it

**Review by ndr-brt — APPROVED** [review]

File: extensions/control-plane/store/sql/contract-negotiation-store-sql/src/main/java/org/eclipse/edc/connector/store/sql/contractnegotiation/SqlContractNegotiationStoreExtension.java [hunk 4]
```diff
@@ -33,9 +34,11 @@
 @Extension(value = "SQL contract negotiation store")
 public class SqlContractNegotiationStoreExtension implements ServiceExtension {
 
-    private static final String DATASOURCE_NAME_SETTING = "edc.datasource.contractnegotiation.name";
     private static final String DEFAULT_DATASOURCE_NAME = "contractnegotiation";
 
+    @Setting(value = "Name of the datasource used to access contract negotiation store", defaultValue = DEFAULT_DATASOURCE_NAME)
```
Comment (jimmarino) [inline extensions/control-plane/store/sql/contract-negotiation-store-sql/src/main/java/org/eclipse/edc/connector/store/sql/contractnegotiation/SqlContractNegotiationStoreExtension.java:39]: I would describe this as: "Name of the datasource used by SQL-based contract negotiation stores"

File: data-protocols/dsp/dsp-api-configuration/src/main/java/org/eclipse/edc/protocol/dsp/api/configuration/DspApiConfigurationExtension.java [hunk 5]
```diff
@@ -37,25 +38,23 @@
...
 @Provides({ DspApiConfiguration.class })
 public class DspApiConfigurationExtension implements ServiceExtension {
-    
+
     public static final String NAME = "Dataspace Protocol API Configuration Extension";
-    
     public static final String CONTEXT_ALIAS = "protocol";
-    
     public static final String DEFAULT_DSP_CALLBACK_ADDRESS = "http://localhost:8282/api/v1/dsp";
+    @Setting(value = "The address where the connector is configured to receive callbacks from the DSP API", defaultValue = DEFAULT_DSP_CALLBACK_ADDRESS)
```
Comment (jimmarino) [inline data-protocols/dsp/dsp-api-configuration/src/main/java/org/eclipse/edc/protocol/dsp/api/configuration/DspApiConfigurationExtension.java:45]: I would describe this as "The endpoint address used by the runtime to receive DSP callbacks"

**Comment by juliapampus** [comment]
Will be reopened later

**Pull request closed without merging**